### PR TITLE
Rewrite ServerTimestamp as a property wrapper

### PR DESCRIPTION
**Pull request by wilhuff**
Add Swift4ServerTimestamp that behaves as before.

### CODE REVIEW

File: Firestore/Swift/Source/Codable/ServerTimestamp.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+#if swift(>=5.1)
+
+  /// A marker protocol that indicates the type can be annotated with an
+  /// `@ServerTimestamp` property wrapper. Only `Timestamp`, `Date`, and `NSDate`
+  /// can be wrapped with `@ServerTimestamp`.
+  public protocol ServerTimestampWrappable {}
+
+  extension Date: ServerTimestampWrappable {}
+  extension NSDate: ServerTimestampWrappable {}
+  extension Timestamp: ServerTimestampWrappable {}
```
Comment (wilhuff) [inline Firestore/Swift/Source/Codable/ServerTimestamp.swift]: @ryanwilson I'm mostly interested in your take on this kind of type constraint. The rest of this is essentially just like ExplicitNull so feel free to ignore.

Comment (ryanwilson) [reply Firestore/Swift/Source/Codable/ServerTimestamp.swift]: I think it's reasonable, and looks like a good solution. One thing that may be helpful is to actually encode what we're trying to get out of the type and implement it in the extension, making the decode a bit cleaner. Potentially just having two properties: seconds and nanoseconds. What do you think?

Comment (wilhuff) [reply Firestore/Swift/Source/Codable/ServerTimestamp.swift]: That's a great idea. It makes the mechanism completely extensible. I've added a test showing how @ServerTimestamp apply to arbitrary types in 4121604.

Comment (ryanwilson) [reply Firestore/Swift/Source/Codable/ServerTimestamp.swift]: Looks great!

**Pull request merged**